### ISSUE TITLE
Definition of "is an array index" makes "-0" an array index

### ISSUE THREAD

**Issue by foolip** [heycam/webidl#516]
https://heycam.github.io/webidl/#dfn-array-index-property-name

This links to https://tc39.github.io/ecma262/#sec-canonicalnumericindexstring which special-cases "-0".

This problem was introduced with https://github.com/heycam/webidl/pull/427, the previous definition was OK:
https://web.archive.org/web/20170903233050/heycam.github.io/webidl/#dfn-array-index-property-name

If the current definition were used, it would mean that `document.all["-0"]` should behave like `document.all["0"]`, which it doesn't as implemented.

**Comment by foolip** [comment]
@TimothyGu can you take a look?

**Comment by TimothyGu** [comment]
I'm inclined to just say "If *P* is an array index" as ES does it, but since that's not preferred here just adding a `"-0"` as an early return is fine.

**Comment by annevk** [comment]
Well, this affects more than just `document.all`, right? I think we very much want the Ecma array index definition for other indexed getters...

**Comment by foolip** [comment]
Yes, it affects more things, and I don't think @TimothyGu was suggested a `document.all`-specific workaround.
